Dockerfile.ci

```diff
@@ -1,12 +1,29 @@
 FROM dependabot/dependabot-core
-ARG CODE_DIR=/home/dependabot/dependabot-core
+
+USER root
+
+ARG DEPENDABOT_HOME=/home/dependabot
+ARG USERNAME=dependabot
+
+# Install project dependencies as an unprivileged user
+RUN useradd -m ${USERNAME} && mkdir ${DEPENDABOT_HOME}/dependabot-core
+RUN chown -R ${USERNAME}:${USERNAME} \
+  ${DEPENDABOT_HOME}/dependabot-core \
+  /usr/local/.pyenv \
+  /opt/go/gopath \
+  /opt/rust/ \
+  /opt/bundler/
+
+ARG CODE_DIR=${DEPENDABOT_HOME}/dependabot-core
 WORKDIR ${CODE_DIR}
 
-ENV BUNDLE_PATH="/home/dependabot/.bundle" \
-  BUNDLE_BIN=".bundle/binstubs" \
-  PATH=".bundle/binstubs:$PATH:/home/dependabot/.bundle/bin"
+USER ${USERNAME}
```

Comment:
This setup is the same as Dockerfile.development and the updater, previously ci would run as root having access to `/var/lib` which is the default cache directory for rubygems when `GEM_HOME` isn't set